apps/marginfi-v2-ui/src/components/common/Portfolio/hooks/use-reward-simulation.tsx

```
@@ -0,0 +1,174 @@
+        atas.push(ata);
+
+        const originData = await marginfiClient.provider.connection.getAccountInfo(ata);
+        if (!originData) continue;
+
+        const beforeAmount = AccountLayout.decode(originData.data).amount.toString();
+        beforeAmounts.set(bankAddress, { amount: beforeAmount, tokenSymbol, mintDecimals });
+      }
+
+      if (beforeAmounts.size === 0) {
+        setSimulationResult({
+          totalReward: 0,
+          rewards: [],
+        });
+        return;
+      }
+
+      const previewAtas = await marginfiClient.simulateTransactions([actionTxn], atas);
+      if (!previewAtas[0]) return;
+
+      previewAtas.forEach((ata, index) => {
+        if (!ata) return;
+
+        const afterAmount = AccountLayout.decode(ata).amount.toString();
+        const bankAddress = bankAddressesWithEmissions[index];
+        const beforeData = beforeAmounts.get(bankAddress);
+
+        if (beforeData) {
+          afterAmounts.set(bankAddress, {
+            amount: afterAmount,
+            tokenSymbol: beforeData.tokenSymbol,
+            mintDecimals: beforeData.mintDecimals,
+          });
+        }
+      });
+
+      let rewards: rewardsType = {
+        totalReward: 0,
+        rewards: [],
+      };
+
+      beforeAmounts.forEach((beforeData, bankAddress) => {
+        const afterData = afterAmounts.get(bankAddress);
+
+        if (afterData) {
+          const beforeAmount = nativeToUi(beforeData.amount, beforeData.mintDecimals);
+          const afterAmount = nativeToUi(afterData.amount, afterData.mintDecimals);
+          const rewardAmount = afterAmount - beforeAmount;
+
+          if (rewardAmount > 0) {
+            rewards.rewards.push({
+              bank: beforeData.tokenSymbol,
+              amount: rewardAmount < 0.01 ? "<0.01" : numeralFormatter(rewardAmount),
+            });
+            rewards.totalReward += rewardAmount;
+          }
+        }
+      });
+
+      setSimulationResult(rewards);
+    } catch (error) {
```

improve error catching, same response as above

---

This should fail silently, i've added in a sentry capture and a console.error but we wont display this to the user, it will just seem like they have no rewards.